client/client.cpp

```diff
@@ -186,10 +192,6 @@ bool Client::receive(std::vector<std::string>& updates) {
     error_ = "No active connection";
     return false;
   }
-  if (!conn_->poll(30000)) {
-    error_ = "Timeout while waiting for server";
-    return false;
-  }
```

Comment:
Well, I would disagree here. In previous code all the send / receive operations (init, send, receive) were blocking, i.e. for them timeout_ms = -1. So to be consistent we should keep -1.

Poll operation was a hack reducing the probability of getting stuck forever, but I would say that in general it must be the end user's responsibility to set the positive timeout value with the meaning of "a rather long wait".